paymentsheet/src/test/java/com/stripe/android/paymentsheet/CustomerSessionPaymentSheetActivityTest.kt

```diff
@@ -0,0 +1,422 @@
+        )
+    }
+
+    private fun createViewModel(
+        arguments: PaymentSheetContractV2.Args,
+        paymentMethods: List<PaymentMethod>,
+        isPaymentMethodRemoveEnabled: Boolean,
+    ): PaymentSheetViewModel = runBlocking {
+        TestViewModelFactory.create(
+            linkConfigurationCoordinator = mock<LinkConfigurationCoordinator>().stub {
+                onBlocking { getAccountStatusFlow(any()) }.thenReturn(flowOf(AccountStatus.SignedOut))
+                on { emailFlow } doReturn stateFlowOf("email@email.com")
+            },
+        ) { linkHandler, linkInteractor, savedStateHandle ->
+            val application = ApplicationProvider.getApplicationContext<Application>()
+            val customer = PaymentSheetFixtures.EMPTY_CUSTOMER_STATE
+
+            PaymentSheetViewModel(
+                application = application,
+                args = arguments,
+                eventReporter = mock(),
+                paymentSheetLoader = FakePaymentSheetLoader(
+                    stripeIntent = PaymentIntentFixtures.PI_REQUIRES_PAYMENT_METHOD,
+                    customer = customer.copy(
+                        paymentMethods = paymentMethods,
+                        permissions = customer.permissions.copy(
+                            canRemovePaymentMethods = isPaymentMethodRemoveEnabled,
+                        )
+                    ),
+                    isGooglePayAvailable = true,
+                    linkState = null,
+                    delay = 0.seconds,
+                    paymentSelection = null,
+                    cbcEligibility = CardBrandChoiceEligibility.Eligible(
+                        preferredNetworks = arguments.config.preferredNetworks,
+                    ),
+                ),
+                customerRepository = FakeCustomerRepository(paymentMethods),
+                prefsRepository = FakePrefsRepository(),
+                logger = Logger.noop(),
+                workContext = UnconfinedTestDispatcher(),
+                savedStateHandle = savedStateHandle,
+                linkHandler = linkHandler,
+                linkConfigurationCoordinator = linkInteractor,
+                intentConfirmationHandlerFactory = IntentConfirmationHandler.Factory(
+                    intentConfirmationInterceptor = FakeIntentConfirmationInterceptor(),
+                    savedStateHandle = savedStateHandle,
+                    stripePaymentLauncherAssistedFactory = object : StripePaymentLauncherAssistedFactory {
+                        override fun create(
+                            publishableKey: () -> String,
+                            stripeAccountId: () -> String?,
+                            statusBarColor: Int?,
+                            includePaymentSheetNextHandlers: Boolean,
+                            hostActivityLauncher: ActivityResultLauncher<PaymentLauncherContract.Args>
+                        ): StripePaymentLauncher {
+                            return mock()
```

Comment:
Same with all of these, we don't expect these to be called. It seems like it would be nice to use the real implementation with `NetworkRule`, or throw here.